File: SDKs/amazonaws.com/iotanalytics/2017-11-27/python/src/sdk/models/shared/channelactivity.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class ChannelActivity:
    r"""ChannelActivity
    The activity that determines the source of the messages to be processed.
    """
    
    channel_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('channelName') }})
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    next: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    

File: SDKs/amazonaws.com/iotanalytics/2017-11-27/java/lib/src/main/java/openapisdk/models/shared/ChannelActivity.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ChannelActivity
 * The activity that determines the source of the messages to be processed.
**/
public class ChannelActivity {
    @JsonProperty("channelName")
    public String channelName;
    public ChannelActivity withChannelName(String channelName) {
        this.channelName = channelName;
        return this;
    }
    @JsonProperty("name")
    public String name;
    public ChannelActivity withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("next")
    public String next;
    public ChannelActivity withNext(String next) {
        this.next = next;
        return this;
    }
}